src/server/game/Spells/Spell.cpp

```diff
@@ -4340,12 +4345,19 @@ void Spell::WriteCastResultInfo(WorldPacket& data, Player* caster, SpellInfo con
         {
              uint32 item = 0;
              for (int8 eff = 0; eff < MAX_SPELL_EFFECTS; eff++)
+             {
                  if (spellInfo->Effects[eff].ItemType)
+                 {
                      item = spellInfo->Effects[eff].ItemType;
-             ItemTemplate const* proto = sObjectMgr->GetItemTemplate(item);
-             if (proto && proto->ItemLimitCategory)
-                 data << uint32(proto->ItemLimitCategory);
-             break;
+                 }
+
+                 ItemTemplate const* proto = sObjectMgr->GetItemTemplate(item);
+                 if (proto && proto->ItemLimitCategory)
+                 {
+                     data << uint32(proto->ItemLimitCategory);
+                 }
+                 break;
+             }
```

Comment:
I dont understand the gcc warning about this

Reply:
@Kitzunu it means you have a `case` inside a `switch` where you don't specify neither `break` nor `[[fallthrough]]``:

```
Spell.cpp:4360:14: error: this statement may fall through
```

https://en.cppreference.com/w/cpp/language/attributes/fallthrough

it's always better to specify `[[fallthrough]]` whenever you did NOT put a `break` INTENTIONALLY. This is meant to avoid bugs where the programmer just forgets to add `break`.